managers/models.py

```diff
@@ -8,8 +12,34 @@ class Project(models.Model):
     start_date = models.DateField()
     stop_date = models.DateField(null=True, blank=True)
     terminated = models.BooleanField(default=False)
-    managers = models.ManyToManyField(CustomUser, related_name="managers")
-    members = models.ManyToManyField(CustomUser, related_name="members")
+    managers = models.ManyToManyField(CustomUser, related_name="managed_projects")
+    members = models.ManyToManyField(CustomUser, related_name="projects")
 
     def __str__(self):
         return self.name
+
+
+@receiver(m2m_changed, sender=Project.managers.through)
+def update_user_type(sender, action, pk_set, **kwargs):
+    project = kwargs['instance']
+    if action == 'pre_remove' or action == 'post_remove':
+        revoke_from_manager(pk_set)
+    elif action == 'pre_add' or action == "post_add":
+        grant_to_manager(project)
+
+
+def grant_to_manager(project):
+    for manager in project.managers.all():
+        if manager.user_type is not CustomUser.UserType.MANAGER.name:
+            manager.user_type = CustomUser.UserType.MANAGER.name
+            manager.full_clean()
+            manager.save()
+
+
+def revoke_from_manager(pk_set):
```

Comment:
I couldn't find any test for this function. Please add them

Reply:
I would also change this name to `change_user_type_to_employee`

Reply:
Done